internal/services/resource/resource_management_private_link_association_resource.go

```diff
@@ -0,0 +1,205 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/resources/2020-05-01/privatelinkassociation"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/resources/2020-05-01/resourcemanagementprivatelink"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var _ sdk.Resource = ResourceManagementPrivateLinkAssociationResource{}
+
+type ResourceManagementPrivateLinkAssociationResource struct{}
+
+func (r ResourceManagementPrivateLinkAssociationResource) ModelObject() interface{} {
+	return &ResourceManagementPrivateLinkAssociationResourceSchema{}
+}
+
+type ResourceManagementPrivateLinkAssociationResourceSchema struct {
+	ManagementGroupId               string `tfschema:"management_group_id"`
+	Name                            string `tfschema:"name"`
+	ResourceManagementPrivateLinkId string `tfschema:"resource_management_private_link_id"`
+	PublicNetworkAccessEnabled      bool   `tfschema:"public_network_access_enabled"`
+	TenantID                        string `tfschema:"tenant_id"`
+}
+
+func (r ResourceManagementPrivateLinkAssociationResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return privatelinkassociation.ValidatePrivateLinkAssociationID
+}
+
+func (r ResourceManagementPrivateLinkAssociationResource) ResourceType() string {
+	return "azurerm_resource_management_private_link_association"
+}
+
+func (r ResourceManagementPrivateLinkAssociationResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			ForceNew:     true,
+			Optional:     true,
+			Computed:     true,
+			Type:         pluginsdk.TypeString,
+			ValidateFunc: validation.IsUUID,
+		},
```

Comment:
Given that this is only used as part of the resource id, would it make sense to remove this entirely and rely on the generated uuid here? In any case, as outlined in the contributors guide [here](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/best-practices.md#setting-properties-to-optional--computed) we are no longer adding Optional and Computed to new fields.

Reply:
If move `name` entirely, will get test error:
```
=== CONT  TestAccResourceManagementPrivateLinkAssociation_requiresImport
    testcase.go:113: Step 2/2, expected an error but got none
```

Reply:
Can I make `name` as required to solve this, but in this way user needs to manually set a uuid for this.

Reply:
Thanks @teowa. On second thought, we should set this to just `Optional` and add a note in the docs that the user should use `ignore_changes` on name if they want to use the generated uuid